numpy/core/function_base.py

```diff
@@ -121,58 +121,82 @@ def linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None,
     if num < 0:
         raise ValueError("Number of samples, %s, must be non-negative." % num)
     div = (num - 1) if endpoint else num
-
-    # Convert float/complex array scalars to float, gh-3504
-    # and make sure one can use variables that have an __array_interface__, gh-6634
```

Comment:
Why revert the code for gh-6634?

Reply:
I didn't revert the code for #6634 or #3504. The original code performs two operations in one statement and this causes a failure for datetime64/timedelta64. So, I had to separate them. There are now two separate code paths: one for datetime64/timedelta64 and one path for all other data types. I simply moved the comment to the appropriate path. Should I change the comments?